ext/mongo_kerberos/mongo_kerberos_native.c

```diff
@@ -104,6 +110,13 @@ static VALUE initialize_challenge(VALUE self) {
   }
 
   context = Data_Wrap_Struct(rb_cObject, NULL, mongo_sasl_conn_free, conn);
+  /* I'm guessing ruby raises on out of memory condition rather than */
+  /* returns NULL, hence no error checking is needed here? */
+  
+  /* from now on context owns conn */
+  /* since mongo_sasl_conn_free cleans up conn, we should NOT call */
+  /* sasl_dispose any more in this function. */
+  RB_GC_GUARD(context);
```

Comment:
I think this should be moved to below line 120? From what I can tell in [the docs](https://docs.ruby-lang.org/en/2.4.0/extension_rdoc.html#label-Appendix+E.+RB_GC_GUARD+to+protect+from+premature+GC), `RB_GC_GUARD)` only guards the context up to the point where the macro is invoked, and so I think we want this invocation to be below the last usage of context.